### PR TITLE
Add IDeviceTwin interface + IoT Hub Device Twin wrapper on it

### PR DESCRIPTION
**Pull request by kbeaugrand**
# PR for Wrapping IoT Hub 

## What is being addressed

The objective is to wrap the IoT Hub implementation to bring after the IoT Central feature.

## How is this addressed

- Adding the ``IoTHubDeviceTwin`` class that inherit from IDeviceTwin, and wraps the IoT Hub Device twin
- Change method signatures to ``IDeviceRegistryManager`` to use this interface
- Refactors the code and unit tests

### CODE REVIEW

File: LoRaEngine/modules/LoRaWanNetworkSrvModule/LoraTools/IoTHubImpl/DeviceTwinExtensions.cs [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+namespace LoRaTools.IoTHubImpl
+{
+    using System;
+    using Microsoft.Azure.Devices.Shared;
+
+    internal static class DeviceTwinExtensions
+    {
+        internal static Twin ToIoTHubDeviceTwin(this IDeviceTwin twin)
+        {
+            ArgumentNullException.ThrowIfNull(twin, nameof(twin));
```
Comment (Mandur) [inline LoRaEngine/modules/LoRaWanNetworkSrvModule/LoraTools/IoTHubImpl/DeviceTwinExtensions.cs:13]: As discussed orally would be nice to have some unit tests here

**Comment by kbeaugrand** [comment]
@danigian and @bastbu, 

Can we merge this PR ?

**Comment by danigian** [comment]
> @danigian and @bastbu,
> 
> Can we merge this PR ?

I just started a final full run of CI and as soon as it passes, I will merge it.
Thanks again for the contribution @kbeaugrand 👍

**Comment by kbeaugrand** [comment]
> 

Thanks @danigian !
More PR are coming after this ;)

**Pull request merged**